src/node_file.h

```diff
@@ -51,25 +66,118 @@ class FSReqWrap : public ReqWrap<uv_fs_t> {
   const char* data_ = nullptr;
   MaybeStackBuffer<char> buffer_;
 
+  DISALLOW_COPY_AND_ASSIGN(FSReqBase);
+};
+
+class FSReqWrap : public FSReqBase {
+ public:
+  FSReqWrap(Environment* env, Local<Object> req)
+      : FSReqBase(env, req, AsyncWrap::PROVIDER_FSREQWRAP) { }
+
+  void FillStatsArray(const uv_stat_t* stat) override;
+  void Reject(Local<Value> reject) override;
+  void Resolve(Local<Value> value) override;
+  void ResolveStat() override;
+  void SetReturnValue(const FunctionCallbackInfo<Value>& args) override;
+
+ private:
   DISALLOW_COPY_AND_ASSIGN(FSReqWrap);
 };
 
+class FSReqPromise : public FSReqBase {
+ public:
+  explicit FSReqPromise(Environment* env);
+
+  ~FSReqPromise() override;
+
+  void FillStatsArray(const uv_stat_t* stat) override;
+  void Reject(Local<Value> reject) override;
+  void Resolve(Local<Value> value) override;
+  void ResolveStat() override;
+  void SetReturnValue(const FunctionCallbackInfo<Value>& args) override;
+
+ private:
+  bool finished_ = false;
+  AliasedBuffer<double, v8::Float64Array> stats_field_array_;
```

Comment:
Can we avoid this in some way, i.e. create the array buffer + the typed array only once we’re actually at the resolve point and know that this is actually a `stat` call?

Reply:
That's essentially what it was, then the change landed today that altered the signature for FillStatsArray to require an AliasedBuffer. I updated today just to get it working again but plan on revisiting before this lands.